backend/webapi/Features/Banners/Index.cs

```diff
@@ -0,0 +1,59 @@
+namespace Pidp.Features.Banners;
+
+using Microsoft.EntityFrameworkCore;
+using NodaTime;
+using Pidp.Data;
+
+public class Index
+{
+    public class Query : IQuery<List<Model>>
+    {
+        public string Component { get; set; } = string.Empty;
+    }
+
+    public class Model
+    {
+        public int Id { get; set; }
+        public string Component { get; set; } = string.Empty;
+        public string Status { get; set; } = string.Empty;
+        public string Header { get; set; } = string.Empty;
+        public string Body { get; set; } = string.Empty;
+        public Instant StartTime { get; set; }
+        public Instant EndTime { get; set; }
+
+    }
+
+
+    public class QueryHandler(PidpDbContext context) : IQueryHandler<Query, List<Model>>
+    {
+        private readonly PidpDbContext context = context;
+
+        public async Task<List<Model>> HandleAsync(Query query)
+        {
+            Console.WriteLine("query component : ");
+            Console.WriteLine(query.Component);
+            var banners = await this.context.Banners
+                .Where(banner => banner.Component == query.Component)
+                .Select(banner => new Model {
+                    Id = banner.Id!,
+                    Component = banner.Component,
+                    Status = banner.Status,
+                    Header = banner.Header,
+                    Body = banner.Body!,
+                    StartTime = banner.StartTime!,
+                    EndTime = banner.EndTime!
+                })
+                .ToListAsync();
+
+            Console.WriteLine(banners.ToArray());
+            var currentTime = DateTime.UtcNow;
+            var unixTime = ((DateTimeOffset)currentTime).ToUnixTimeMilliseconds();
+
+            var activeBanners = banners.Where(banner => banner.StartTime.ToUnixTimeMilliseconds() < unixTime &&
+                    banner.EndTime.ToUnixTimeMilliseconds() > unixTime
+                ).ToList();
```

Comment:
Two bits of feedback here:
1. Please use `NodaTime` for date operations like this. You can inject `IClock` into the handler and use `clock.GetCurrentInstant()` to get the current time and compare it directly to `StartTime` and `EndTime` without having to do conversions.
2. This `.Where` clause should be in the DB query for performance and readability reasons. The code as written is fetching every Banner from the DB and then filtering them down, which isn't necessary. You can add the conditionals to the `.Where` on line 36.

Reply:
fixed.